worker/src/direct_invocation/direct_invocation_ocall.rs

```diff
@@ -0,0 +1,152 @@
+        !matches!(
+            status,
+            TrustedOperationStatus::Invalid
+                | TrustedOperationStatus::InSidechainBlock(_)
+                | TrustedOperationStatus::Finalized
+                | TrustedOperationStatus::Usurped
+        )
+    }
+}
+
+impl<W> DirectInvocationBridge for DirectInvocationOCall<W>
+where
+    W: WatchListFacade,
+{
+    fn update_status_event(
+        &self,
+        hash_vec: Vec<u8>,
+        status_update_vec: Vec<u8>,
+    ) -> OCallBridgeResult<()> {
+        let status_update = TrustedOperationStatus::decode(&mut status_update_vec.as_slice())
+            .map_err(|e| OCallBridgeError::DirectInvocationError(format!("{:?}", e)))?;
+
+        let hash = Hash::decode(&mut hash_vec.as_slice())
+            .map_err(|e| OCallBridgeError::DirectInvocationError(format!("{:?}", e)))?;
+
+        let continue_watching = match self.watch_list.get_watching_client(&hash) {
+            Some(watched_client) => {
+                let old_result: Vec<u8> = watched_client.response.result.clone();
+                let mut result = RpcReturnValue::decode(&mut old_result.as_slice()).unwrap();
+
+                let do_watch = DirectInvocationOCall::<W>::continue_watching(&status_update);
+
+                // update response
+                result.do_watch = do_watch;
+                result.status = DirectRequestStatus::TrustedOperationStatus(status_update);
+
+                let mut updated_client = watched_client;
+                updated_client.response.result = result.encode();
+
+                updated_client
+                    .client
+                    .send(serde_json::to_string(&updated_client.response).unwrap())
+                    .unwrap();
+
+                // update entire watching client object
+                self.watch_list.add_watching_client(hash, updated_client);
+
+                do_watch
+            }
+            None => false,
+        };
+
+        if !continue_watching {
+            self.watch_list.remove_watching_client(&hash);
+        }
+
+        Ok(())
+    }
+
+    fn send_status(&self, hash_vec: Vec<u8>, status_vec: Vec<u8>) -> OCallBridgeResult<()> {
+        let hash = Hash::decode(&mut hash_vec.as_slice())
+            .map_err(|e| OCallBridgeError::DirectInvocationError(format!("{:?}", e)))?;
+
+        if let Some(watched_client) = self.watch_list.get_watching_client(&hash) {
+            let client = &watched_client.client;
+            let mut response = watched_client.response.clone();
+
+            // create return value
+            // TODO: Signature?
+            let submitted =
+                DirectRequestStatus::TrustedOperationStatus(TrustedOperationStatus::Submitted);
+            let result = RpcReturnValue::new(status_vec, false, submitted);
+
+            // update response
+            response.result = result.encode();
+
+            client
+                .send(serde_json::to_string(&response).unwrap())
+                .unwrap();
```

Comment:
I'd vote for getting rid of these unwraps, preferably with `?` or `map_err`

At some other places in this PR too. We repeatedly suffer from the worker server being killed due to a panic in the direct invocation code.